pkg/storage/gcp/bucket_cloud.go

```diff
@@ -0,0 +1,78 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"github.com/gomods/athens/pkg/config"
+	s "github.com/gomods/athens/pkg/storage"
+	multierror "github.com/hashicorp/go-multierror"
+	"google.golang.org/api/iterator"
+)
+
+type gcpBucket struct {
+	*storage.BucketHandle
+}
+
+func (b *gcpBucket) DeleteModule(ctx context.Context, module, version string) error {
```

Comment:
Does GCP Storage have nesting capabilities? If so, an enhancement would be to nest the three "objects" below within one path and so you don't have to do 3 roundtrips just to delete a module, you can just delete the parent path. Doesn't have to be in this pr of course.